flax/experimental/nnx/nnx/graph_utils.py

```diff
@@ -152,6 +153,9 @@ def is_node(x: tp.Any) -> bool:
 def is_node_type(x: type[tp.Any]) -> bool:
   return x in NODE_TYPES or x is PytreeType
 
+def is_state_leaf(x: tp.Any) -> bool:
```

Comment:
Can this be a `typing.TypeGuard`?

Reply:
great idea! I should start using `TypeGuard` more